### PR TITLE
Don't allow concurrent pstacks

### PR DESCRIPTION
**Pull request by markhannum**
Simple flag allows only a single pstack at a time.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>2/555 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_downgrade<br>truncatesc_offline_generated</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>3/555 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>lostwrite [setup failure]<br>truncatesc<br>truncatesc_offline_generated</p>

File: bdb/file.c [hunk 1]
```diff
@@ -2196,9 +2196,12 @@ static void bdb_appsock(netinfo_type *netinfo, SBUF2 *sb)
 }
 
 extern int gbl_pstack_self;
+static int running_pstack = 0;
 void pstack_self(void)
 {
-    if (!gbl_pstack_self) return;
+    if (!gbl_pstack_self || running_pstack)
```
Comment (riverszhang89) [inline bdb/file.c:2202]: since this is to prevent concurrent pstack callers, does this check need to be protected by a mutex?

Comment (markhannum) [reply bdb/file.c]: > New version protects this under a mutex

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ bdb/file.c @@
+ Please make sure that a mutex is unlocked before returning.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/555 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>op_makerecord<br>lostwrite<br>sc_downgrade<br>truncatesc_offline_generated</p>

**Review by aakash10292 — APPROVED** [review]

**Pull request merged**